iothub/device/tests/Transport/AmqpIot/AmqpIotErrorAdapterTests.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Net.WebSockets;
+using FluentAssertions;
+using Microsoft.Azure.Amqp;
+using Microsoft.Azure.Amqp.Framing;
+using Microsoft.Azure.Devices.Client.Transport.AmqpIot;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Azure.Devices.Client.Tests
+{
+    [TestClass]
+    [TestCategory("Unit")]
+    public class AmqpIotErrorAdapterTests
+    {
+        [TestMethod]
+        public void AmqpIotErrorAdapter_ExceptionFromOutcome_Null()
+        {
+            Exception ex = AmqpIotErrorAdapter.GetExceptionFromOutcome(null);
+            ex.Should().BeEquivalentTo(new IotHubClientException("Unknown error."));
+        }
+
+        [TestMethod]
+        public void AmqpIotErrorAdapter_ExceptionFromOutcome_Rejected()
+        {
+            Exception ex = AmqpIotErrorAdapter.GetExceptionFromOutcome(new Rejected());
+            ex.Should().BeEquivalentTo(new IotHubClientException("Unknown error."));
+        }
+
+        [TestMethod]
+        public void AmqpIotErrorAdapter_ExceptionFromOutcome_Released()
+        {
+            Exception ex = AmqpIotErrorAdapter.GetExceptionFromOutcome(new Released());
+            ex.Should().BeEquivalentTo(new OperationCanceledException("AMQP link released."));
+        }
+
+        [TestMethod]
+        [DataRow("amqp:internal-error", IotHubClientErrorCode.NetworkErrors)]
```

Comment:
Out of scope of this PR, but should we map internal error to something like ServerError?

Reply:
That would make more sense, yes.

Reply:
I'm not sure I'd agree, @abhipsaMisra. I'd take AMQP internal error to be about the AMQP library, not a response from the service. Do you know something about that error in particular?

Reply:
You're right, in this context the internal error would indicate an error form the AMQP library - either from the client side or service-side. ServerError wouldn't be the best choice or this scenario.
However, I wonder if marking this as transient network error is the best solution here. The [amqp specs](http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-transport-v1.0-os.html#type-amqp-error) indicate that this error _might_ require manual intervention.
I'm ok with us waiting until someone reports an issue with this error though. This isn't something we've seen people encounter in the past.

<html>
<body>
<!--StartFragment-->

internal-error |   | amqp:internal-error
-- | -- | --
An internal error occurred. Operator intervention might be necessary to resume normal operation.

<!--EndFragment-->
</body>
</html>